src/cr/cube/dimension.py

```diff
@@ -591,11 +591,17 @@ def _element_dicts(self):
 
     def _element_id_from_dict(self, element_dict):
         """Returns the element identifier given its dict."""
+        selector = self._elements_transforms.get("selector")
+        default_id = element_dict["id"]
         if self._dimension_type in DT.ARRAY_TYPES:
-            # --- In case of array dimensions returns the id of the subvariable.
-            return element_dict.get("value", {}).get("id")
+            element_value = element_dict.get("value", {})
+            if selector == "alias":
+                # --- In case of specified key alias, returns the subvariable alias.
+                return element_value.get("references", {}).get("alias") or default_id
+            # --- Returns the subvariable id.
```

Comment:
Honestly, I'd prefer an `elif` here, just to be more explicit about the fact that we're waiting for a `"subvariable_id"` rather than `"fake_selector"`. Maybe even raise on `else` with the `ValueError("Unknown selector type %s" % selector)`

Reply:
oh I cannot do the `ValueError`. Consider that now we do not support key spec in the frontend so what we receive today is `{"S2": {"hide": True}}` without the key spec. We need a fallback case. In future this can be implemented but now is not feasible IMHO